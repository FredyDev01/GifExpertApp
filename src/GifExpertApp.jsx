import { AddCategory, RemoveCategory, GifGrid } from "./components"
import { useLocalStorage } from "./hooks/useLocalStorage"


export const GifExpertApp = () => {   
    
    const [categories, setCategories] = useLocalStorage()
    
    const onAddCategory = (newCategory)=> {
        if(categories.length && categories.find(e => e.toUpperCase() == newCategory.toUpperCase())) return;        
        setCategories([newCategory, ...categories])
    }

    const onRemoveCategory = (DeleteAll)=> {               
        setCategories(DeleteAll ? [] : categories.filter((e, id)=> id != 0), true)             
    }

    return (
        <>
            <header>
                <h1>GifExpertApp</h1>
                <AddCategory onNewCategory={value => onAddCategory(value) }/>
                <RemoveCategory onDeleteCategory={value => onRemoveCategory(value)}/>
            </header>
            <ol>
                {
                    categories.length ?  
                    categories.map(categ => (
                        <GifGrid key={categ} categ={categ}/>
                    )) 
                    :
                    <p className="infoData">Al parecer no tiene ninguna categoria de busqueda registrada.</p>
                }
            </ol>
            
        </>
    )
}