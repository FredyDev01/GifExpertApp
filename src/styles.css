* {
    font-family: Helvetica, Arial, sans-serif;    
  }

  body{
    margin: 0px;
    background: #f1f1f1;
  }

  header {
    padding: 60px;
    background-image: linear-gradient(to left, rgba(94, 112, 245, 0.85), rgba(94, 112, 245, 0.85)), url("/FondoHeader.gif");
    background-position: center center;
    background-size: cover;
    background-attachment: fixed;    
  }

  input {
    background-color: white;
    border-radius: 5px;
    border: 2px solid rgb(221, 225, 233);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 12px 15px;
    width: 100%;
  }

  h1 {
    font-size: 3rem;
    text-align: center;
    color: white;
  }
  
  h3 {
    margin-top: 90px;
    font-size: 1.8rem;
    font-weight: bold;
    display: table;        
  }

  h3 hr{
    margin: 5px 0px 0px 0px;
    width: 60%;
    border: 2px solid rgba(94, 112, 245);
    border-radius: 10rem;
  }

  ol{
    margin: 0px;
    padding: 0px 60px 90px 60px;
  }

  .ContentOptions{
    margin: 30px auto;
    display: table;
  }

  .ContentOptions button{
    background: transparent;
    margin: 0px 8px;
    padding: 8px 12px;
    border: 2px solid white;
    border-radius: 5px;
    cursor: pointer;
    transition: 200ms;
    color: white;
  }

  .ContentOptions button:hover{
    background: white;
    color: black;
  }

  .card-grid {
    margin-top: 75px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card, .infoData{
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  }

  .card { 
    max-width: 350px;
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;    
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }

  .infoData {
    margin: 30px auto 0px auto;
    background: white;
    border-radius: 5px;
    display: table;
    padding: 15px 18px;    
    font-size: 14px;
    text-align: center;
  }