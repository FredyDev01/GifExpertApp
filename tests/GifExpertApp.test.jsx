import { fireEvent, getAllByRole, getByRole, render, renderHook, screen } from "@testing-library/react"
import {GifExpertApp} from '../src/GifExpertApp'


describe('Pruebas en <GifExpertApp />', () => {
    const FirstCategory = 'Isekai ojisan'

    test('debe de hacer match en el snapshot', ()=> {
        const {container} = render(<GifExpertApp />)
        expect(container).toMatchSnapshot()
    })

    test('debe de agregar nuevas categorias', ()=> {
        render(<GifExpertApp />)
        const input = screen.getByRole('searchbox')
        const form = screen.getByRole('form')        
        fireEvent.input(input, {target: {value: FirstCategory}})
        fireEvent.submit(form)
        expect(screen.getByText(FirstCategory)).toBeTruthy()
    })

    test('debe de eliminar la busqueda mas reciente', ()=> {
        render(<GifExpertApp />)
        const input = screen.getByRole('searchbox')
        const form = screen.getByRole('form')        
        fireEvent.input(input, {target: {value: FirstCategory}})
        fireEvent.submit(form)
        fireEvent.click(screen.getByText('Borrar busqueda'))        
        expect(screen.queryByText(FirstCategory)).toBeNull()
    })

    test('debe de eliminar todas las busquedas', ()=> {
        render(<GifExpertApp />)
        const input = screen.getByRole('searchbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, {target: {value: FirstCategory}})
        fireEvent.submit(form)
        fireEvent.click(screen.getByText('Borrar todo'))                        
        expect(screen.queryAllByRole('heading', {level: 3}).length).toBe(0)
    })
})