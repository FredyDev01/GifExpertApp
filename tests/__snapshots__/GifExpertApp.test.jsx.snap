// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de hacer match en el snapshot 1`] = `
<div>
  <header>
    <h1>
      GifExpertApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar gifs :D"
        type="search"
        value=""
      />
    </form>
    <div
      class="ContentOptions"
    >
      <button
        class="BtnDel"
      >
        Borrar busqueda
      </button>
      <button
        class="BtnDelAll"
      >
        Borrar todo
      </button>
    </div>
  </header>
  <ol>
    <h3>
      Demon slayer
       
      <hr />
    </h3>
    <h2>
      Cargando...
    </h2>
  </ol>
</div>
`;
